includes/class-newspack-ads-model.php

```diff
@@ -199,9 +207,6 @@ public static function update_ad_unit( $ad_unit ) {
 	 * @param integer $id The id of the ad unit to delete.
 	 */
 	public static function delete_ad_unit( $id ) {
-		if ( ! current_user_can( 'unfiltered_html' ) ) {
-			return false;
-		}
 		$ad_unit_post = \get_post( $id );
```

Comment:
Not a new issue from this PR, but while we're hardening the logic we should make sure the post type is ad unit so people can't delete arbitrary posts.

Reply:
This should do it: https://github.com/Automattic/newspack-ads/pull/15/commits/a9eaf915a22dc0e085161d7b9e3017dc431f5709